src/crypto/mnemonic/mnemonic.cpp

```diff
@@ -31,4 +47,28 @@ namespace mnemonic
                 });
         return "";
     }
-}
\ No newline at end of file
+
+    WordList entropy2Mnemonic(const std::vector<uint8_t>& entropy, const language::Dictionary& dict)
+    {
+        std::vector<uint8_t> hash(CSHA256::OUTPUT_SIZE);
+
+        CSHA256 sha256;
+        sha256.Write(entropy.data(), entropy.size()).Finalize(hash.data());
+
+        auto checksumBits = entropy.size() * 8 / 32;
+        std::string checksumString = binaryString(hash.data(), (checksumBits/8)+1).substr(0, checksumBits);
+        std::string entString = binaryString(entropy.data(), entropy.size()) + checksumString;
+
+        assert(entString.size() % 11 == 0); // getting 11 bits to pick index in 0-2047
+        assert(entString.size() / 11 == MNEMONIC_WORD_COUNT); // maybe this doesn't have to be constant?
+
+        std::vector<int> inds(MNEMONIC_WORD_COUNT);
+        for(auto i = 0; i < entString.size() / 11; i++) {
+            inds[i] = std::stoi(entString.substr(11*i, 11), nullptr, 2);
```

Comment:
Hmm, is there a particular reason this needs to be a num to string and back to num conversion? Can't you just shift and mask to pull the index out?